Utils/Dataflow/017_adjustMetadata/adjustMetadata.py

```diff
@@ -224,7 +220,9 @@ def process(stage, message):
     if inp_events:
         data['input_events'] = inp_events
     # 5. Save chain_data as array of integers, extract chain_id from it
-    transform_chain_data(data)
+    e = transform_chain_data(data)
+    if e:
+        sys.stderr.write(e)
```

Comment:
Er, I actually meant something like
```
err_pattern = "(ERROR) Processing failed for taskid='%s': %%s.\n" % data.get('taskid', 'unknown')
e = transform_chain_data(data)
if e:
    sys.stderr.write(err_pattern % e)
```

So that `process()` took care of "how to output message ID" etc.

And actually it was a stupid thing to suggest: the proper way to pass error message back to the caller would be to throw an exception :( sorry about it...

Why an exception:
* exception can carry the error message;
* function can still return True/False or whatever;
* exceptions can be of different types, and the caller may decide basing on it if "everything bad, stop processing!" or "something\`s wrong, but we still can keep going".

Reply:
Reverted the changes with returning messages to `process()`(in another branch, see above). Now, when `process()` checks type and taskid, and `transform_chain_data()` does its own checks, should be enough for now. Will rework latter to exceptions if necessary.